packages/core/src/request-builder/header-builder/auth-headers.ts

```diff
@@ -0,0 +1,162 @@
+    error => {
+      throw errorWithCause(
+        'Request for "OAuth2ClientCredentials" authentication access token failed or denied.',
+        error
+      );
+    }
+  );
+  return `Bearer ${response.access_token}`;
+}
+
+function headerFromBasicAuthDestination(destination: Destination): string {
+  if (isNullish(destination.username) || isNullish(destination.password)) {
+    throw Error(
+      'AuthenticationType is "BasicAuthentication", but "username" and / or "password" are missing!'
+    );
+  }
+
+  return basicHeader(destination.username, destination.password);
+}
+
+export function basicHeader(username: string, password: string): string {
+  return `Basic ${Buffer.from(`${username}:${password}`).toString('base64')}`;
+}
+
+function headerForPrincipalPropagation(destination: Destination): MapType<any> {
+  const principalPropagationHeader =
+    destination?.proxyConfiguration?.headers?.[
+      'SAP-Connectivity-Authentication'
+    ];
+  if (!principalPropagationHeader) {
+    throw Error(
+      'Principal propagation was selected in destination, but no SAP-Connectivity-Authentication bearer header was added by connectivity service.'
+    );
+  }
+  return {
+    'SAP-Connectivity-Authentication': principalPropagationHeader
+  };
+}
+
+function headerForProxy(destination: Destination): MapType<any> {
+  const proxyAuthHeader =
+    destination?.proxyConfiguration?.headers?.['Proxy-Authorization'];
+
+  return proxyAuthHeader ? { 'Proxy-Authorization': proxyAuthHeader } : {};
+}
+
+function legacyNoAuthOnPremiseProxy(destination: Destination): MapType<any> {
+  logger.warn(
+    `You are using \'NoAuthentication\' in destination: ${destination.name} which is an OnPremise destination. This is a deprecated configuration, most likely you wanted to set-up \'PrincipalPropagation\' so please change the destination property to the desired authentication scheme.`
+  );
+
+  let principalPropagationHeader;
+  try {
+    principalPropagationHeader = headerForPrincipalPropagation(destination);
+  } catch (e) {
+    logger.warn('No principal propagation header found.');
+  }
+
+  // TODO the proxy header are for OnPrem auth and are now handled correctly in the authorization-header.ts and should be removed here
+  // However this would be a breaking change, since we recommended to use 'NoAuthentication' to achieve principal propagation as a workaround.
+  return {
+    ...headerForProxy(destination),
+    ...principalPropagationHeader
+  };
+}
+
+function getProxyRelatedAuthHeaders(destination: Destination): MapType<any> {
+  if (
+    destination.proxyType === 'OnPremise' &&
+    destination.authentication === 'NoAuthentication'
+  ) {
+    return legacyNoAuthOnPremiseProxy(destination);
+  }
+
+  // The connectivity service will raise an exception if it can not obtain the 'Proxy-Authorization' and the destination lookup will fail early
+  return headerForProxy(destination);
+}
+
+async function getAuthenticationRelatedAuthHeaders(
```

Comment:
I find the name a bit tautologish (I do not know if this is the adjective for Tautology).  You have auth headers for the Proxy and auth headers for the target system. Perhaps call it like this: `DestinationRelatedAuthHeaders` or `TargetSystemRelatedAuthHeaders`....